src/main/scala/org/zalando/spearheads/innkeeper/services/PathsService.scala

```diff
@@ -57,17 +58,40 @@ class DefaultPathsService @Inject() (pathsRepo: PathsRepo)(implicit val executio
 
     pathsRepo.pathWithUriHostIdExists(path.uri, path.hostIds)
       .flatMap {
-        case false => pathsRepo.insert(pathRow).flatMap(rowToEventualMaybePath)
-        case true  => Future.successful(Failure(DuplicatePathUriHost()))
+        case false =>
+          val insertResult = pathsRepo.insert(pathRow)
+
+          insertResult.onSuccess {
+            case insertedPath => insertedPath.id.foreach { id =>
+              auditsRepo.persistPathLog(id, createdBy.name, AuditType.Create)
+            }
+          }
+
+          insertResult.flatMap(rowToEventualMaybePath)
+
+        case true =>
+          Future.successful(Failure(DuplicatePathUriHost()))
       }
   }
 
   override def patch(
     id: Long,
     path: PathPatch,
+    userName: String,
     updatedAt: LocalDateTime): Future[ServiceResult.Result[PathOut]] = {
 
-    pathsRepo.update(id, path, updatedAt).flatMap {
+    val updateResult = pathsRepo.update(id, path, updatedAt)
+
+    updateResult.onSuccess {
```

Comment:
Here are the alternatives:
1. DB triggers
2. At a Repository level

Seeing how the first option was dropped in order to avoid logic at a DB level, the only option left is the second one. I chose to do it at a service level because a service is the place I see this kind of logic belong. I could not abstract this any further. The data available at the service is required in order to create the audit entries. I'm open for suggestions.